philips_hue.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import requests
+import json
+from dotenv import load_dotenv
+
+load_dotenv()
+URL = os.environ.get("philips-url")
+
+
+# sets the light with light_id to the given hue
+def set_color(light_id, hue):
+    r = requests.put(url = f'{URL}/lights/{light_id}/state', data=json.dumps({"on": True, "hue": hue}))
+    print(r.content)
+
+
+# turns Philips Hue red if incident is open, green if closed
+def configure_light(response, light_id):
```

Comment:
I think that we need to be careful about mixing the business logic with the HTTP logic. On one hand, there are functions that will simply be thin wrappers around HTTP endpoints. On the other hand, there will be functions specific to the business logic of your app. I think the API could be improved by keeping these separate (so then it's easy to change one without changing the other). I think it also makes the intent or goal of each function clearer to the caller.

I think that there are multiple ways to go about this. Maybe:
   1. Move configure_light() out of this module.
   2. Or, keep the business logic in this module. But create a Client() object. So then then the api could be:

   client = phillips_hue.Client(url, light_id)
   phillips_hue.configure_light(client)

Or both, or some third option. Those are just some suggestions, I would experiment to see what makes the most sense.

Reply:
Got it, that makes sense. I ended up just moving configure_light() to main since it calls set_color() in the philips_hue file.